gbpservice/neutron/tests/unit/nfp/configurator/agents/test_vpn_agent.py

```diff
@@ -0,0 +1,68 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+
+from gbpservice.nfp.configurator.agents import vpn
+from gbpservice.nfp.configurator.drivers.vpn.vyos import vyos_vpn_driver
+from gbpservice.neutron.tests.unit.nfp.configurator.test_data import (
+                                                                test_vpn_data)
+import unittest
+import mock
+
+"""
+Implements test cases for RPC manager methods of vpn agent
+"""
+
+
+class VPNaasEventHandlerTestCase(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super(VPNaasEventHandlerTestCase, self).__init__(*args, **kwargs)
+        self.dict_obj = test_vpn_data.MakeDictionaries()
+        self.handler = vpn.VPNaasEventHandler(self.dict_obj.sc,
+                                              self.dict_obj.drivers)
+        self.ev = test_vpn_data.FakeEvent()
+        self.driver = vyos_vpn_driver.VpnaasIpsecDriver(
+                                                self.handler.plugin_rpc)
+
+    def test_handle_event(self):
+        '''
+        Test to handle the vpn agent's vpnservice_updated method to
+        handle various vpn operations
+
+        '''
+        with mock.patch.object(self.handler,
```

Comment:
handle_event() function handles both VYOS_VPN_SYNC and VPNSERVICE_UPDATED events. Where is the testcase for the sync? This testcase seems to be testing only VPNSERVICE_UPDATED event.